DuckDuckGo/AppConfigurationFetch.swift

```diff
@@ -78,9 +86,62 @@ class AppConfigurationFetch {
             } else {
                 UIApplication.shared.endBackgroundTask(taskID)
             }
+
             completion?(newData)
         }
     }
+
+    @available(iOS 13.0, *)
+    static func registerBackgroundRefreshTaskHandler() {
+        BGTaskScheduler.shared.register(
+            forTaskWithIdentifier: AppConfigurationFetch.Constants.backgroundProcessingTaskIdentifier,
+            using: fetchQueue) { (task) in
+
+            task.expirationHandler = {
+                scheduleBackgroundRefreshTask()
+            }
+
+            AppConfigurationFetch().fetchConfigurationFiles(isBackground: true)
+            scheduleBackgroundRefreshTask()
+
+            task.setTaskCompleted(success: true)
+        }
+    }
+
+    @available(iOS 13.0, *)
+    static func scheduleBackgroundRefreshTask() {
+        let task = BGProcessingTaskRequest(identifier: AppConfigurationFetch.Constants.backgroundProcessingTaskIdentifier)
+        task.requiresNetworkConnectivity = true
+        task.earliestBeginDate = Date(timeIntervalSinceNow: Constants.minimumConfigurationRefreshInterval)
+
+        // Background tasks can be debugged by breaking on the `submit` call, stepping over, then running the following LLDB command, before resuming:
+        //
+        // e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateLaunchForTaskWithIdentifier:@"com.duckduckgo.app.configurationRefresh"]
+
+        do {
+            try BGTaskScheduler.shared.submit(task)
+        } catch {
+            os_log("Error when submitting background task: %s", log: generalLog, type: .debug, error.localizedDescription)
```

Comment:
Question: do we want to report this via `Pixel`? This can throw when we've reached the maximum task limit (10), or if the background task is configured incorrectly. The current set of statistics are enough to detect issues with the new background fetch behaviour, so logging should be fine, but I'm open to changes.

Reply:
I'd add debug pixel here (along with error code), we can always get rid of it after validating outcome of this project. If the expectation is that this should not happen, then it's easy to check - number of pixels should be 0. :)